src/Libraries/CoreNodes/Data.cs

```diff
@@ -51,16 +70,114 @@ private static object ToNative(JToken token)
             }
         }
 
+        /// <summary>
+        /// Parse implementation for converting JObject types to specific Dynamo objects (ie Geometry, Color, Images, etc) 
+        /// </summary>
+        /// <param name="jObject"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        private static object DynamoJObjectToNative(JObject jObject)
+        {
+            if (jObject.ContainsKey("$typeid"))
+            {
+                var typeid = jObject["$typeid"].ToString();
+
+                switch (typeid)
+                {
+                    //autodesk.math to abstract ProtoGeometry types
+                    case "autodesk.math:vector3d-1.0.0":
+                        return Vector.FromJson(jObject.ToString());
+
+                    case "autodesk.math:matrix44d-1.0.0":
+                        return CoordinateSystem.FromJson(jObject.ToString());
+
+                    //autodesk.geometry to abstract ProtoGeometry types
+                    case "autodesk.geometry:boundingbox3d-1.0.0":
+                        return BoundingBox.FromJson(jObject.ToString());
+
+                    case "dynamo.geometry:mesh-1.0.0":
+                        return Mesh.FromJson(jObject.ToString());
+
+                    //types supported by Goemetry.FromJson
+                    case "autodesk.math:point3d-1.0.0":
+                    case "dynamo.geometry:sab-1.0.0":
+                    case "dynamo.geometry:tsm-1.0.0":
+                    case "dynamo.geometry:rectangle-1.0.0":
+                    case "dynamo.geometry:cuboid-1.0.0":
+                    case "dynamo.geometry:solid-1.0.0":
+                    case string geoId when geoId.Contains("autodesk.geometry"):
+                        return Geometry.FromJson(jObject.ToString());
+
+                    //Dynamo types
+                    case "dynamo.graphics:color-1.0.0":
+                        return Color.ByARGB(
+                        (int)jObject["A"],
+                        (int)jObject["R"],
+                        (int)jObject["G"],
+                        (int)jObject["B"]);
+#if _WINDOWS
+                    case "dynamo.graphics:png-1.0.0":
+                        jObject.TryGetValue(ImageFormat.Png.ToString(), out var value);
+
+                        if (value != null)
+                        {
+                            var stream = Convert.FromBase64String(value.ToString());
+
+                            Bitmap bitmap;
+                            using (var ms = new MemoryStream(stream))
+                                bitmap = new Bitmap(Bitmap.FromStream(ms));
+
+                            return bitmap;
+                        }
+#endif
+
+                        return null;
+
+                    case "dynamo.data:location-1.0.0":
+                        return DynamoUnits.Location.ByLatitudeAndLongitude(
+                        (double)jObject["Latitude"],
```

Comment:
curious - what are your thoughts about this vs adding protected or internal json constructors and just letting newtonsoft figure it out? 

I'm fine with this for now, just wanted to evaluate the +/-s

Reply:
I think that is better approach.  I was waiting for the .NET7 so we could introduce an interface to support serialization registration similar to the IGraphicItem.  (static abstract is now supported in interfaces NET7 and above).  I can do this now without the interface for the internal types just to set the pattern.

Reply:
ah I see, so we'll have to wait on the interface approach until after we move to .net 8... I guess you could do it now and change the language target of this csproj to 11?